src/core/utils/Debounce/Debounce.tsx

```diff
@@ -0,0 +1,49 @@
+import { Component } from 'react';
+
+export type DebounceProps<T> = T & {
+  waitFor: number;
+};
+export class Debounce<T extends Object> extends Component<DebounceProps<T>> {
+  timeout: ReturnType<typeof setTimeout> | null = null;
+
+  componentWillUnmount() {
+    this.clearTimeout();
+  }
+
+  debouncer = (callback: Function, value?: T) => {
+    if (!callback) {
+      return;
+    }
+
+    if (this.props.waitFor === 0) {
+      callback(value);
+      return;
+    }
+
+    this.clearTimeout();
+    const { waitFor } = this.props;
+    this.timeout = setTimeout(() => {
+      callback(value);
+    }, waitFor);
+  };
+
+  cancelDebounce = () => {
+    this.clearTimeout();
+  };
+
+  clearTimeout() {
```

Comment:
Would it possible to have different name for this method?
It is now little confusing to read, when the `clearTimeout` is called and it will call the JS "internal" one and not this.

So `this.clearTimeout` and `clearTimeout` are too close to each other when looking this.

Reply:
Changed it to cancelDebounce, which also allowed me to delete the extra method passed as a render prop.

Reply:
I take that back somewhat. Losing the middle man method somehow seemed to break things, so I just renamed the internal method to cancelTimeout.

Reply:
Good catch, should be changed in my opinion too!